### PR TITLE
Eng 529 db connection restore

### PR DESCRIPTION
**Pull request by firegloves**
Added some logic to populate 4 env vars useful for wildfly to reconnect automatically to the DB after connection lost.
They are used into entando-docker-base-images > entando-wildfly12-base

Env vars are:

PORTDB_CONNECTION_CHECKER
PORTDB_EXCEPTION_SORTER
SERVDB_CONNECTION_CHECKER
SERVDB_EXCEPTION_SORTER

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **werbth**

If they are not already assigned, you can assign the PR to them by writing `/assign @werbth` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando-k8s%2Fentando-k8s-app-controller).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando-k8s/entando-k8s-app-controller/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-jx** [comment]
@firegloves: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | b071f3fe0ff5b50152be6765aacce70dfcb08fb9 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-k8s-app-controller-acpr8**

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-k8s-app-controller-acpr8**

File: src/main/java/org/entando/kubernetes/controller/app/JbossDatasourceValidation.java
Comment (Kerruba) [inline src/main/java/org/entando/kubernetes/controller/app/JbossDatasourceValidation.java:26]: Is this only a Jboss thing or generic?

File: src/test/java/org/entando/kubernetes/controller/app/inprocesstests/DeployEntandoServiceTest.java
Comment (Kerruba) [inline src/test/java/org/entando/kubernetes/controller/app/inprocesstests/DeployEntandoServiceTest.java]:
```suggestion
                "mysql", "3306", "my_app_portdb");
```

File: src/test/java/org/entando/kubernetes/controller/app/inprocesstests/DeployEntandoServiceTest.java
Comment (Kerruba) [inline src/test/java/org/entando/kubernetes/controller/app/inprocesstests/DeployEntandoServiceTest.java]:
```suggestion
                "mysql", "3306", "my_app_servdb");
```

File: src/test/java/org/entando/kubernetes/controller/app/testutils/EnvVarAssertionHelper.java
Comment (Kerruba) [inline src/test/java/org/entando/kubernetes/controller/app/testutils/EnvVarAssertionHelper.java]:
```suggestion
                is("org.jboss.jca.adapters.jdbc.extensions.mysql.MySQLValidConnectionChecker"));
```

File: src/test/java/org/entando/kubernetes/controller/app/testutils/EnvVarAssertionHelper.java
Comment (Kerruba) [inline src/test/java/org/entando/kubernetes/controller/app/testutils/EnvVarAssertionHelper.java]:
```suggestion
                is("org.jboss.jca.adapters.jdbc.extensions.mysql.MySQLExceptionSorter"));
```

File: src/test/java/org/entando/kubernetes/controller/app/inprocesstests/DeployEntandoDbConsumerTest.java
Comment (Kerruba) [inline src/test/java/org/entando/kubernetes/controller/app/inprocesstests/DeployEntandoDbConsumerTest.java]:
```suggestion
                "postgresql", "5432", "my_app_db");
```

File: src/test/java/org/entando/kubernetes/controller/app/inprocesstests/DeployEntandoDbConsumerTest.java
Comment (Kerruba) [inline src/test/java/org/entando/kubernetes/controller/app/inprocesstests/DeployEntandoDbConsumerTest.java]:
```suggestion
                "postgresql", "5432", "my_app_db");
```

**Review by Kerruba — CHANGES_REQUESTED** [review]
Let's try to use strings so that we are sure that when things are changing the the ENUM the test is going to break

File: src/main/java/org/entando/kubernetes/controller/app/JbossDatasourceValidation.java
Comment (firegloves) [reply src/main/java/org/entando/kubernetes/controller/app/JbossDatasourceValidation.java:26]: Yes it is only Jboss related. The referenced classes are jboss internals

**Comment by entando-jx** [comment]
:star: PR built and available in a preview environment **entando-k8s-app-controller-acpr8**

**Review by Kerruba — APPROVED** [review]
Looks good to me

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando-k8s/entando-k8s-app-controller/pull/8#pullrequestreview-444795016" title="Approved">Kerruba</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando-k8s%2Fentando-k8s-app-controller).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando-k8s/entando-k8s-app-controller/blob/master/OWNERS)~~ [Kerruba]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
